pinot-timeseries/pinot-timeseries-spi/src/main/java/org/apache/pinot/tsdb/spi/series/builders/MaxTimeSeriesBuilder.java

```diff
@@ -39,14 +40,15 @@ public MaxTimeSeriesBuilder(String id, TimeBuckets timeBuckets, List<String> tag
 
   @Override
   public void addValueAtIndex(int timeBucketIndex, Double value) {
-    if (_values[timeBucketIndex] == null || value > _values[timeBucketIndex]) {
-      _values[timeBucketIndex] = value;
+    Double expectedValue = (value == null) ? (Double) 0.0 : value;
```

Comment:
Using 0.0 as default is incorrect behavior imo because you can have negative and null values together.

If the existing value is null, we can simply return. (similar comment for other builders)

Reply:
Modified the code to do nothing if incoming value is null.